ffi/src/sspi/sec_handle.rs

```diff
@@ -1212,3 +1220,274 @@ pub extern "system" fn QueryCredentialsAttributesExW(
 }
 
 pub type QueryCredentialsAttributesExFnW = extern "system" fn(PCredHandle, u32, *mut c_void, u32) -> SecurityStatus;
+
+#[cfg(test)]
+mod tests {
+    use std::ffi::CStr;
+    use std::ptr::{null, null_mut};
+
+    use libc::c_void;
+
+    use crate::sspi::common::{DeleteSecurityContext, FreeContextBuffer, FreeCredentialsHandle};
+    use crate::sspi::sec_buffer::{SecBuffer, SecBufferDesc};
+    use crate::sspi::sec_handle::{
+        AcquireCredentialsHandleA, AcquireCredentialsHandleW, InitializeSecurityContextA, InitializeSecurityContextW,
+        SecHandle,
+    };
+    use crate::sspi::sec_pkg_info::{
+        EnumerateSecurityPackagesA, EnumerateSecurityPackagesW, PSecPkgInfoA, PSecPkgInfoW, QuerySecurityPackageInfoA,
+        QuerySecurityPackageInfoW, SecPkgInfoA, SecPkgInfoW,
+    };
+    use crate::sspi::sec_winnt_auth_identity::{SecWinntAuthIdentityA, SecWinntAuthIdentityW};
+    use crate::utils::c_w_str_to_string;
+
+    extern "system" fn dummy(_: *mut c_void, _: *mut c_void, _: u32, _: *mut *mut c_void, _: *mut i32) {}
+
+    #[test]
+    fn initialize_security_context_w() {
+        // This test simulates initialize security context function call. It's better to run it using Miri
+        // https://github.com/rust-lang/miri
+        // cargo +nightly miri test
+        let mut pkg_name = "NTLM\0".encode_utf16().collect::<Vec<_>>();
+        let mut pkg_info: PSecPkgInfoW = null_mut::<SecPkgInfoW>();
+
+        let status = unsafe { QuerySecurityPackageInfoW(pkg_name.as_ptr(), &mut pkg_info) };
+        assert_eq!(status, 0);
+
+        println!("{:?}", unsafe { &*pkg_info });
```

Comment:
I left those `println!`s on purpose: just to simulate any memory access to the allocated memory.

Reply:
Sounds good! Can you write this down as an in-code comment please?

Reply:
done